### DISCUSSION TITLE
Adding artificial sink reactions

### DISCUSSION THREAD

**Discussion by manas-kohli** [SysBioChalmers/Human-GEM#693] [Q&A] [answered]
Hey there,

This may seem like a silly question but is there a simple function within RAVEN or Human1 to add sink or demand reactions since these were removed from the original Human1 model? Like for example, if I wanted to add a sink reaction for intracellular cholesterol, how would I go about doing that. Many thanks in advance!

**Comment by mihai-sysbio** [comment]
Great question @manas-kohli! And it sounds to me like a nice thing to add in the [Human-GEM guide](https://sysbiochalmers.github.io/Human-GEM-guide/), especially since this can be achieved also with other toolboxes besides RAVEN.

**Comment by manas-kohli** [comment]
Hi @mihai-sysbio thanks for your answer. When you mention 'also with other toolboxes' firstly do you mean you could do this with RAVEN because if so, that would be the simplest. Otherwise there is an addsinkReactions function in COBRA so would I need to use the RAVEN-Cobra wrapper function to convert my models to the cobra format followed by adding the sink reaction and evaluating the flux in cobra instead of RAVEN?

**Reply by mihai-sysbio** [reply]
From what I can see there is no simple function in RAVEN that would add everything back (like in COBRA), specially that it's been a long time since [this removal](https://github.com/SysBioChalmers/Human-GEM/pull/90) took place and the identifiers have changed in the meantime, so even the [preserved list of removed reactions](https://github.com/SysBioChalmers/Human-GEM/blob/main/.deprecated/data/modelCuration/removedSinkDMrxns.tsv) isn't of immediate use.
However, since you are asking particularly about a single metabolite I'm wondering if it isn't as simple as adding one or several reactions manually and using[`setExchangeBounds`](https://github.com/SysBioChalmers/RAVEN/blob/da86f6246b27b1d347850375d718933842077732/core/setExchangeBounds.m#L27).

**Comment by JonathanRob** [comment, marked as answer]
Hi @manas-kohli, I don't know of a function that would automatically add a sink reaction for the metabolite of interest, though one could add it in a somewhat more manual fashion using RAVEN's `addRxns` function; e.g.

```
rxnStruct = {};
rxnStruct.rxns = {'sink_cholesterol'};
rxnStruct.equations = {'cholesterol[c] =>'};

newModel = addRxns(model, rxnStruct, 3);
```

Alternatively, depending on if your solver allows, one can use the `b` field in the model after converting it to 2 columns (representing max depletion and accumulation, respectively), and changing the b-value for the metabolites for which you would like to add "sinks". For example:

```
model.b = zeros(numel(model.mets), 2);
sinkMet = getIndexes(model, 'cholesterol[c]', 'mets');
model.b(sinkMet, 2) = 1000:
```

This will allow cholesterol[c] to be produced by other reactions in the model with a net flux of up to 1000 (mmol/gDW h) without anything consuming it (i.e., it "accumulates"), which is effectively the same as adding a sink reaction with max flux of 1000.

Probably not a bad idea to experiment with both to verify that it is doing what you expect.

**Comment by manas-kohli** [comment]
Thanks, Jon and Mihail, I'll try this out now

**Comment by manas-kohli** [comment]
Hi Jon, 

Both approaches are doing more or less what I expect and the values I'm getting for my ec_models seem to be fine. I'm trying to see whether this approach can give me a rough estimate of relative levels of ability to produce metabolites and how closely this is to actual measured data so I'll perform this for more metabolites and see what results I get. Many thanks for your prompt help, I really appreciate it.
